apps/api/src/app/bridge/usecases/preview-step/preview-step.usecase.ts

```diff
@@ -10,44 +10,47 @@ export class PreviewStep {
   constructor(private executeBridgeRequest: ExecuteBridgeRequest) {}
 
   async execute(command: PreviewStepCommand): Promise<ExecuteOutput> {
-    const event = this.mapEvent(command);
-
-    const response = (await this.executeBridgeRequest.execute(
-      ExecuteBridgeRequestCommand.create({
-        environmentId: command.environmentId,
-        action: PostActionEnum.PREVIEW,
-        event,
-        searchParams: {
-          [HttpQueryKeysEnum.WORKFLOW_ID]: command.workflowId,
-          [HttpQueryKeysEnum.STEP_ID]: command.stepId,
-        },
-        // TODO: pass the origin from the command
-        workflowOrigin: WorkflowOriginEnum.EXTERNAL,
-        retriesLimit: 1,
-      })
-    )) as ExecuteOutput;
+    const event = this.buildBridgeEventPayload(command);
+    const executeCommand = this.createExecuteCommand(command, event);
+
+    const bridgeResult = await this.executeBridgeRequest.execute(executeCommand);
+
+    return bridgeResult as ExecuteOutput;
+  }
 
-    return response;
+  private createExecuteCommand(command: PreviewStepCommand, event: Event) {
+    return ExecuteBridgeRequestCommand.create({
+      environmentId: command.environmentId,
+      action: PostActionEnum.PREVIEW,
+      event,
+      searchParams: {
+        [HttpQueryKeysEnum.WORKFLOW_ID]: command.workflowId,
+        [HttpQueryKeysEnum.STEP_ID]: command.stepId,
+      },
+      // TODO: pass the origin from the command
+      workflowOrigin: WorkflowOriginEnum.NOVU_CLOUD,
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** We need to derive the `workflowOrigin` from the `command`, this implementation will break existing external Bridge workflow previews.‏